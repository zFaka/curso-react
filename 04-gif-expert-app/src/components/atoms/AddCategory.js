import React, {useState} from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {

  const [textareaValue, setTextareaValue] = useState('');

  const handleTextareaChange = (inputTxt) => {
    setTextareaValue(inputTxt.target.value)
  };

  const handleSubmmit = (submmitedValue) => {

    submmitedValue.preventDefault();

    if (textareaValue.trim().length > 1) {
      setCategories(oldTextareaValue=>[textareaValue, ...oldTextareaValue]);
      setTextareaValue('');
    }

  };

  return (
    <form onSubmit={handleSubmmit}>
      {/*<p>{textareaValue}</p>*/}
      <input 
        type='text' 
        placeholder='Any idea?'
        value={textareaValue} 
        onChange={handleTextareaChange}
      />
    </form>
  )}

AddCategory.propTypes = {setCategories: PropTypes.func.isRequired}
