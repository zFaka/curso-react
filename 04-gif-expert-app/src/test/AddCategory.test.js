
import {shallow} from 'enzyme';
import React from 'react';
import {AddCategory} from '../components/atoms/AddCategory';

describe('Testing AddCategory', () => {

  const setCategories = jest.fn();
  let wrapper = shallow(<AddCategory setCategories={setCategories}/>);

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={setCategories}/>);
  })

  test('Should return a func', () => {

    expect(wrapper).toMatchSnapshot()})

  //test('Should change textarea', () => {

    //const input = wrapper.find('input');
    //const value = 'loquesea';
    //input.simulate('change', {target: {value:value}});

    //expect(wrapper.find('p').text().trim()).toBe(value)

  //})

  test('Should not post info with submit', () => {

    wrapper.find('form').simulate('submit', { preventDefault(){} });

    expect(setCategories).not.toHaveBeenCalled();

  })

  test('Should call setCategories and clean textarea', () => {

    const value = 'Jake';

    // inputChange simulation
    wrapper.find('input').simulate('change', {target:{value:value}}); 

    // submit simulation
    wrapper.find('form').simulate('submit', {preventDefault(){}});

    // call setCategories
    expect(setCategories).toHaveBeenCalled();
    expect(setCategories).toHaveBeenCalledTimes(1);
    expect(setCategories).toHaveBeenCalledWith(expect.any(Function));

    // clean inputValue 
    expect(wrapper.find('input').prop('value')).toBe('');

  })

})
