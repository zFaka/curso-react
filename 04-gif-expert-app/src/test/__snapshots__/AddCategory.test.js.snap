// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing AddCategory Should return a func 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Any idea?"
    type="text"
    value=""
  />
</form>
`;
